src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/InputOverloadMethodsDeclarationOrderRecords.java

```diff
@@ -60,7 +60,7 @@ public void foo() { // violation
         public MyClass() {
         }
 
-        public void foo(int i, String s) {
+        public void foo(int i, String s) { // violation
         }
 
         public void foo(String s, int i) {
```

Comment:
Why isn't this a violation?

Reply:
it is a violation. I've added the `// violation` comment to it.

Reply:
> it is a violation. I've added the `// violation` comment to it to indicate that's a violation

CI is green and no changes have been pushed up, so I am not sure what this comment means.

Reply:
> I am not sure what this comment means.

This was the new violation introduced due to the changes in the check implementation. I've added the violation comment and handled it here:

https://github.com/checkstyle/checkstyle/blob/922cba38624e05652b2d9cb37f13e342d0d4c03b/src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java#L62

So there's no CI failing and everything is working fine as expected.

Reply:
This comment is for line 66, which is why I have placed it there.

Reply:
Sorry, but I'm not able to understand what do you mean by line 66. 

https://github.com/checkstyle/checkstyle/blob/922cba38624e05652b2d9cb37f13e342d0d4c03b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/coding/overloadmethodsdeclarationorder/InputOverloadMethodsDeclarationOrderRecords.java#L57-L67

As you can see here, there is a class between the `foo()` & `foo(int i, String s)`. That's why there is a violation at `foo(int i, String s)` indicating that it should be placed next to `foo()` i.e the class should not be between the overloaded methods. There is no violation for line 66 because there is the overloaded `foo(int i, String s)` above it.

Reply:
So to sum it up, we just check if one overloaded method is sitting against another. Not that all overloaded methods need to be moved to one spot.

In the long run, both mean the same thing, but in a single slice of time, it is a difference. 66 would be a violation if only line 63 was moved. However, technically line 57 could be moved to 63 and everything would be compliant in terms of this check.

Reply:
Ah, so a user would have to run multiple checkstyle executions to make this check happy. I see.